### PR TITLE
Feature/ukr 215 update registration flow

### PR DESCRIPTION
**Pull request by dannewnsjump24**
This brings in a new component for the registration flow to handle the type of orders the merchant can receive.

There are frontend tests for the new component as well as updated tests for the backend process of saving the data so all the post request tests have been updated to show options with the new table service method selected.

### CODE REVIEW

File: resources/js/components/shared/business_details/MerchantOrderTypes.vue [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  data() {
+    return {
+      showDeliveryFields: false,
+    };
+  },
+  methods: {
+    showDeliveryData() {
+      this.showDeliveryFields = !this.showDeliveryFields;
+    },
+    doesOrderTypeExist(collectionType) {
```
Comment (AlexJump24) [inline resources/js/components/shared/business_details/MerchantOrderTypes.vue:135]: I think we need to add some unit tests against this method.

File: resources/js/bootstrap.js [hunk 2]
```diff
@@ -49,12 +51,13 @@ orderFilters.init();
 const adminMenu = new AdminMenu();
 adminMenu.init();
 
-Vue.config.devtools = false;
+Vue.config.devtools = true;
```
Comment (AlexJump24) [inline resources/js/bootstrap.js:54]: Renabled

Comment (dannewnsjump24) [reply resources/js/bootstrap.js]: Ive just added the process check to this so it now uses the app env to make sure its only on with `local` environments

File: app/Http/Requests/Registration/BusinessDetailsRequest.php [hunk 3]
```diff
@@ -29,18 +29,18 @@ public function rules()
             'logo' => ['image'],
             'name' => ['required'],
             'description' => ['required', new MaxWordsRule(100)],
-            'url-slug' => ['required', 'alpha_dash', 'unique:merchants,url_slug'],
-            'collection_type' => ['required', Rule::in(['collection', 'both', 'delivery'])],
+            'url_slug' => ['required', 'alpha_dash', 'unique:merchants,url_slug'],
+            'collection_types' => ['required'],
```
Comment (AlexJump24) [inline app/Http/Requests/Registration/BusinessDetailsRequest.php:33]: Do we have tests for any of this validation? I had a look but can't see any just a test that asserts not data was present and asserts that the url slug is present in the session errors. Maybe we could just add collection_type to this test to ensure no-one could ever submit without a collection type?

File: resources/js/components/shared/business_details/MerchantOrderTypes.vue (hunk 1 above)
Comment (dannewnsjump24) [reply resources/js/components/shared/business_details/MerchantOrderTypes.vue:135]: Good idea ill get onto that

File: app/Http/Requests/Registration/BusinessDetailsRequest.php (hunk 3 above)
Comment (dannewnsjump24) [reply app/Http/Requests/Registration/BusinessDetailsRequest.php:33]: so there are tests that cover the post aspects of these routes but im going to add some more work to them particularly around collection_types

File: resources/js/components/shared/business_details/MerchantOrderTypes.vue (hunk 1 above)
Comment (dannewnsjump24) [reply resources/js/components/shared/business_details/MerchantOrderTypes.vue:176]: Ive added in relevant tests now for this

**Pull request merged**